CreateLegalMessage/handler.ts

```diff
@@ -0,0 +1,152 @@
+  RequestHandler
+} from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+import { RequiredParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_param";
+import { Context } from "@azure/functions";
+import { IAzureApiAuthorization } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { EmailString } from "@pagopa/ts-commons/lib/strings";
+import { ImpersonatedService } from "../generated/api-admin/ImpersonatedService";
+import { ErrorResponses } from "../utils/responses";
+import { withApiRequestWrapper } from "../utils/api";
+import { APIClient } from "../clients/admin";
+import { ILogger, getLogger } from "../utils/logging";
+import { ILegalMessageMapModel, notFoundError } from "../utils/legal-message";
+
+const logPrefix = "CreateLegalMessageHandler";
+
+const getImpersonatedService = (
+  logger: ILogger,
+  adminClient: APIClient,
+  serviceId: string
+): TE.TaskEither<ErrorResponses, ImpersonatedService> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      adminClient.getImpersonatedService({
+        serviceId
+      }),
+    200
+  );
+
+type ICreateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  rawRequest: express.Request,
+  legalmail: EmailString
+) => Promise<ErrorResponses | IResponseSuccessJson<ImpersonatedService>>;
+
+/**
+ * Handles requests for imporsonate service by a input serviceId.
+ */
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+export function ImpersonateServiceHandler(
+  adminClient: APIClient,
+  lmMapper: ILegalMessageMapModel
+): ICreateServiceHandler {
+  return (
+    context,
+    _auth,
+    rawRequest,
+    legalmail
+  ): ReturnType<ICreateServiceHandler> => {
+    const bbb = pipe(
+      legalmail,
+      lmMapper.findLastVersionByModelId,
+      TE.chain(
+        TE.fromOption(() =>
+          notFoundError("Can not found a service with the input legal mail")
+        )
+      ),
+      TE.map(lmMap => lmMap.serviceId),
+      TE.mapLeft(e =>
+        e.kind === "NotFoundError"
+          ? ResponseErrorNotFound("Not Found", e.message)
+          : ResponseErrorInternal(e.message)
+      ),
+      TE.chainW(serviceId =>
+        getImpersonatedService(
+          getLogger(context, logPrefix, "ImpersonateService"),
+          adminClient,
+          serviceId
+        )
+      ),
+      TE.map(impersonatedService => {
+        // eslint-disable-next-line functional/immutable-data
+        rawRequest.headers["x-user-groups"] =
+          impersonatedService.user_groups || "";
+        // FIXME: const HEADER_USER_EMAIL = "x-user-email";
+        //        const HEADER_USER_SUBSCRIPTION_KEY = "x-subscription-id";
+        return impersonatedService;
+      }),
+      TE.map(ResponseSuccessJson),
```

Comment:
why not just return the new request instead of mutating the argument? It's not clear why this method should be an handler (since we call CreateMessage handler anyway) and not a simple function that transform the passed in request.